bus-mapping/src/evm/opcodes/call.rs

```diff
@@ -0,0 +1,250 @@
+use super::Opcode;
+use crate::circuit_input_builder::CircuitInputStateRef;
+use crate::eth_types::{GethExecStep, ToWord};
+use crate::operation::{
+    AccountField, AccountOp, CallContextField, CallContextOp, StackOp,
+    TxAccessListAccountOp,
+};
+use crate::{operation::RW, Error};
+
+/// Placeholder structure used to implement [`Opcode`] trait over it
+/// corresponding to the `OpcodeId::DUP*` `OpcodeId`.
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct Call;
+
+impl Opcode for Call {
+    fn gen_associated_ops(
+        state: &mut CircuitInputStateRef,
+        steps: &[GethExecStep],
+    ) -> Result<(), Error> {
+        let tx_id = 1;
+        let caller_call = state.caller_call().clone();
```

Comment:
Sure, will add some comments first.

> what about doing the handling of CALL inside `gen_associated_ops`

I tried this idea, but then we need to pass the `is_success` into `gen_associated_ops`, which then requires change on the trait :(

Another idea just comes to my mind: Could I attach the `is_success` in `CircutiInputBuilderRef`? Then I don't have to change the trait to get the value.

Reply:
> Another idea just comes to my mind: Could I attach the `is_success` in `CircutiInputBuilderRef`? Then I don't have to change the trait to get the value.

You mean the `call_is_success: HashMap` right?  I think it makes sense to move that data into a struct.  And actually this will be required if we do this: https://github.com/appliedzkp/zkevm-circuits/pull/278#discussion_r785305147
I think we can move this task to an issue to keep the PR as it is.  What do you think?

Reply:
> You mean the call_is_success: HashMap right?

Yes, and we will need such info if we want to handle call/create in `gen_associated_ops`.

> I think we can move this task to an issue to keep the PR as it is

Sure, we can try to resolve it in further PRs.